gh-106307: Add PyMapping_GetOptionalItem() (#106308)
python/cpython: Doc/whatsnew/3.13.rst

  ``NULL`` if the referent is no longer live.
  (Contributed by Victor Stinner in :gh:`105927`.)

* Add :c:func:`PyMapping_GetOptionalItem` and

>   :c:func:`PyMapping_GetOptionalItemString` function: replacements of

Reviewer: ```suggestion
  :c:func:`PyMapping_GetOptionalItemString`: variants of
```

  :c:func:`PyObject_GetAttr` and :c:func:`PyMapping_GetItemString` which don't
  raise :exc:`KeyError`.

>   It is convenient and faster if the missing key should not be treated

Reviewer: ```suggestion
  These variants are more convenient and faster if the missing key should not be treated
```
